Cap'n Proto And Fibers / Exceptions

I did a bump of system-installed packages and there may be an issue with some distributions of Cap'n Proto for versions which have fibers enabled. It looks like fibers are incompatible with no-exception build options. E.g., I started to see the following build failure on a rebuild of CIRCT:

```
/nix/store/4w0pgdp0nv391sksvi179zjq3x7wvifz-capnproto-0.10.1/include/kj/async.h:39:6: error: "Fibers cannot be enabled when exceptions are disabled."
    #error "Fibers cannot be enabled when exceptions are disabled."
     ^
1 error generated.
```

This is probably a @teqdruid question on how to handle this.

I can always disable Cap'n Proto in my build.

--- a/include/circt/Dialect/ESI/cosim/CMakeLists.txt
+++ b/include/circt/Dialect/ESI/cosim/CMakeLists.txt
@@ -12,6 +12,12 @@ if(CapnProto_FOUND)
   set(COSIM_SCHEMA_HDR ${CIRCT_BINARY_DIR}/include/circt/Dialect/ESI/CosimSchema.h)
   configure_file(CosimSchema.h.in ${COSIM_SCHEMA_HDR})
 
+  if (MSVC)
+    string(REPLACE "/EHs-c-" "" CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
+  else ()
+    string(REPLACE "-fno-exceptions" "" CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
+  endif ()
+
   add_definitions(${CAPNP_DEFINITIONS})
   capnp_generate_cpp(COSIM_CAPNP_SRCS COSIM_CANPN_HDRS CosimDpi.capnp)
   add_library(EsiCosimCapnp

--- a/lib/Dialect/ESI/cosim/cosim_dpi_server/CMakeLists.txt
+++ b/lib/Dialect/ESI/cosim/cosim_dpi_server/CMakeLists.txt
@@ -10,6 +10,12 @@ if(ESI_COSIM)
     Server.cpp
     Endpoint.cpp)
 
+  if (MSVC)
+    string(REPLACE "/EHs-c-" "" CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
+  else ()
+    string(REPLACE "-fno-exceptions" "" CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
+  endif ()
+
   set_target_properties(EsiCosimDpiServer
       PROPERTIES
           LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib

